include/circt/Dialect/OM/OMTypes.td

```diff
@@ -35,6 +35,15 @@ def ReferenceType : TypeDef<OMDialect, "Reference", []> {
   let mnemonic = "ref";
 }
 
+def ListType : TypeDef<OMDialect, "List", []> {
+  let summary = "A type that represents a list.";
+
+  let mnemonic = "list";
+  let parameters = (ins "mlir::Type":$elementType);
```

Comment:
Probably another design would be to encode the length of elements as a parameter like FIRRTL's vector or HW's array. For object model, i think it sounds reasonable to me not to encode element length but i'm open for suggestions.